Makefile

```diff
@@ -94,26 +94,24 @@ ensure-generate-is-noop: set-image-controller generate bundle
 	@git diff -s --exit-code docs/api.md || (echo "Build failed: the api.md file has been changed but the generated api.md file isn't up to date. Run 'make api-docs' and update your PR." && git diff && exit 1)
 
 .PHONY: all
-all: manager
-.PHONY: ci
-ci: test
+all: manager targetallocator operator-opamp-bridge
 
-# Run tests
-# setup-envtest uses KUBEBUILDER_ASSETS which points to a directory with binaries (api-server, etcd and kubectl)
-.PHONY: test
-test: generate fmt vet ensure-generate-is-noop envtest
-	KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(KUBE_VERSION) -p path)" go test ${GOTEST_OPTS} ./...
+# No lint here, as CI runs it separately
+.PHONY: ci
+ci: generate fmt vet test ensure-generate-is-noop
 
 # Build manager binary
 .PHONY: manager
-manager: generate fmt vet
+manager:
```

Comment:
`generate` generates deepcopy methods for the `api` package wich seems fundamental for  manager binary

Reply:
I don't think there's any reason to run it every time we build the manager, though. We don't do this for other binaries.

Reply:
The majority of contributors don't know what the `generate` command does. There is not even documentation that for the change on the objects in the  `api` package the command should be run.

Reply:
This is actually documented here: https://github.com/open-telemetry/opentelemetry-operator/blob/main/CONTRIBUTING.md#make-changes-to-the-project-manifests.

But in general, I think this is ok. If the contributor should've run `generate`, but didn't, the CI will catch that, as will running `make precommit`.

What I'm trying to achieve with this PR is to not surprise users with changes to files in version control. Running tests or building the operator binary should not do this, in my view, even if it may be convenient at times.

Reply:
The makefile was scaffolded by the operator-sdk e.g. (https://github.com/operator-framework/operator-sdk/blob/master/testdata/go/v4/memcached-operator/Makefile#L145)


I am not sure how this improves dev experience if people will run manger with outdated generate methods.

Reply:
I am fine splitting manifest generation from the compilation. But I would expect that compilation produces up-to-date binaries.

Reply:
I can see that argument. In that case, should I also add `generate` to the opamp bridge make target?

Reply:
For the record, I definitely disagree with running `fmt` as part of this target, so I guess me and operator-sdk are enemies now. 🗡️

Reply:
I think we should be consistent and if opapm requires the generate command it should be added as well.

Reply:
Added `generate` as a dependency of both `manager` and `opamp-bridge`.